### PR TITLE
Script for splitting audio files by silences

### LINKED ISSUE #46: Separate audio by sentence (or silence)
Support module to help separate audio files for training purposes

### LINKED ISSUE #87: Return file cut short when testing autotranslation
When testing autotranslation, I used the following command on a one-minute spanish file cut from the beginning of Angelina.wav and received a five second, one-sentence output instead of the full length audio. The file is inside the zip below as well. 

(I used the one minute file to save time/money but also because the original 25 minute file returns an unrelated error about being too long. )

``` python -m CogNative.main -sampleAudio "I:\github\repo\CogNative\CogNative\examples\AngelinaShort.wav" -synType audio -dialogueAudio "I:\github\repo\CogNative\CogNative\examples\AngelinaShort.wav" -out "I:\github\repo\CogNative\CogNative\examples\AngelinaShortClone.wav" -useExistingEmbed y```

[AngelinaShort.zip](https://github.com/andrew-fennell/CogNative/files/8516979/AngelinaShort.zip)

### PR DESCRIPTION
**Pull request by ctis98**
Closes #46

### CODE REVIEW

File: CogNative/backend/modules/audio_split.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from pydub import AudioSegment
+from pydub.silence import split_on_silence
+from colorama import Fore
+import os
+
+# user input for audio file + file existence check
+aud_file = input("Enter name of audio file: ")
+file_check = os.path.exists(f"Clone_Tests/Raw/{aud_file}.mp3")
+assert file_check, print(f"{Fore.YELLOW}Please move audio file to 'Clone_Tests/Raw/'")
```
Comment (JacobSmith1020) [inline CogNative/backend/modules/audio_split.py]: Might want to change extension to wav and adjust the filepath for future integration.

**Review by JacobSmith1020 — DISMISSED** [review]
Needs some minor changes to be compatible with the current project but it's fine for now as a standalone file.

**Comment by andrew-fennell** [comment]
## Summary of changes made

The script will **ensure** that EVERY chunk is under 45 seconds long (good to send to Google's API).

I added a "optimize" option, but it is really just there to try to encourage the chunks to be closer to 45 seconds. The flow is _might_ be better when optimize is used, but we will need to test it to see.

## Time taken to split
### TCOC40.wav
Initial script: 02:34

New version, unoptimized slices: 02:32

New version, optimized slices:  06:01

### Angelina.wav
Initial  script: 00:32

New version, unoptimized slices: 00:39

New version, optimized slices: 01:27

**Comment by andrew-fennell** [comment]
## Tests to run on this branch

```python -m CogNative.main```

```python -m CogNative.audio_split```

```python -m CogNative.STT```

**Review by JacobSmith1020 — CHANGES_REQUESTED** [review]
@andrew-fennell Walked through CogNative.main, CogNative.backend.modules.STT, and CogNative.backend.modules.audio_split. All working for small file inputs but I ran through main with the TCOC40 for audio transcription and ran into an issue after the transcription was done. It looks like some segment of the translator threw an error for a large transcription.

image

**Review by JacobSmith1020 — APPROVED** [review]
The audio splitting is working great :)

**Comment by andrew-fennell** [comment]
This PR has fixed this issue mentioned in #87 due to the audio splitting.

**Pull request merged**